src/components/personalSettings/profileSettings/profileSettings.jsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+
+import style from './profileSettings.module.css';
+import { useForm } from '../../formValidation/index';
+import { connect } from 'react-redux';
+import { setUserDataThunkCreator } from '../../../redux/auth-reducer';
+
+function ProfileSettings(props) {
+  const initialState = {
+    firstname: '',
+    lastname: '',
+    dateOfBirth: '',
+    country: '',
+    gender: ''
+  };
+  function Submit() {
+    let data = {};
+    if (values.firstname) data.firstname = values.firstname;
+    if (values.lastname) data.lastname = values.lastname;
+    if (values.dateOfBirth) data.dateOfBirth = values.dateOfBirth;
+    if (values.country) data.country = values.country;
+    if (values.gender) data.gender = values.gender;
+    props.setUserDataThunkCreator(data);
+  }
```

Comment:
I suppose you don't need to check for presence every value.
you should be able to just call:     props.setUserDataThunkCreator(values);